postgres/tests/test_statements.py

```diff
@@ -1052,21 +1042,22 @@ def wait(conn):
 
         assert event['ddtags'] == expected_tags
 
-        if POSTGRES_VERSION == '9.5':
-            # rest of test is to confirm blocking behavior
-            # which we cannot collect in pg v9.5 at this time
-            return
-
         # ... now run the check again after closing blocking_bob's conn.
         # this means we should report bob as no longer blocked
         # close blocking_bob's tx
         blocking_conn.close()
+
+        # wait for query to complete, but commit has not been called,
+        # so it should remain open and idle
+        wg.wait()
```

Comment:
This test has been flakey for a long time now (example [here](https://github.com/DataDog/integrations-core/actions/runs/5607901533/job/15192585005)). 

specifically the typical failure was [on this line ](https://github.com/DataDog/integrations-core/blob/master/postgres/tests/test_statements.py#L1078)
```
assert "active" == "idle in transaction"
```

Basically there was a race here:

1. closing the blocking connection should unblock bob's query, which tries to read from the `persons` table
2. we expect the state to be `idle in transaction`, since the query would have executed, but `commit` was not called & so the transaction should still be open

... but _sometimes_ the check would run & catch the query still in the `active` state, so it would fail.

In order to prevent this, we relied on this sleep 
```
time.sleep(dbm_instance['query_activity']['collection_interval'])
```

... instead we can simply use a `WaitGroup` to wait until the query has finished executing so we can be sure what the state will be by the time we run this check.